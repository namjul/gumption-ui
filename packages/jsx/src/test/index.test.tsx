/** @jsx jsx ../ */

import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import { jsx } from '..';

afterEach(cleanup);

describe('otion', () => {
  test('renders', () => {
    const { asFragment } = render(<div>Hello World</div>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('render `style` attribute', () => {
    const { asFragment } = render(
      <div style={{ color: 'red' }}>Hello World</div>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('render `class` attribute', () => {
    const { asFragment } = render(<div className="Hello">World</div>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('interpret `compileStyles` attribute', () => {
    const { asFragment } = render(
      <div
        style={{ color: 'red' }}
        className="let gumption"
        compileStyles={(props) => ({
          ...props,
          className: `${props.className} come to you`,
        })}
      >
        Hello World
      </div>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
