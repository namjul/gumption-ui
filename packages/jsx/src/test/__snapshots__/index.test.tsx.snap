// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`otion interpret \`compileStyles\` attribute 1`] = `
<DocumentFragment>
  <div
    class="let gumption come to you"
    style="color: red;"
  >
    Hello World
  </div>
</DocumentFragment>
`;

exports[`otion render \`class\` attribute 1`] = `
<DocumentFragment>
  <div
    class="Hello"
  >
    World
  </div>
</DocumentFragment>
`;

exports[`otion render \`style\` attribute 1`] = `
<DocumentFragment>
  <div
    style="color: red;"
  >
    Hello World
  </div>
</DocumentFragment>
`;

exports[`otion renders 1`] = `
<DocumentFragment>
  <div>
    Hello World
  </div>
</DocumentFragment>
`;
